redis/pom.xml

```diff
@@ -42,4 +54,18 @@ LICENSE file.
       <scope>provided</scope>
     </dependency>
   </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.5.1</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
```

Comment:
so the expectation here is that the non-cluster redis client will also not work with jdk7? can we break this into a different module so that the prior support for jdk7 remains?